### ISSUE TITLE
Evaluation of NER model is producing incorrect results

### ISSUE THREAD

**Issue by erotavlas** [explosion/spaCy#3224]
**How to reproduce the behaviour**

using training script https://github.com/explosion/spaCy/blob/master/examples/training/train_ner.py

I executed this command on the train_ner.py script with my training data (approximately 1000 sentences)

`python train_ner.py -m en_core_web_lg -o outputModel`

**Your Environment**

Operating System:
Windows 10

**Python Version Used:**
Anaconda environment with python version 3.6.8

**spaCy Version Used:**
2.0.18

**Evaluation Script**

```
import spacy
from spacy.gold import GoldParse
from spacy.scorer import Scorer

def evaluate(ner_model, examples):
    scorer = Scorer()
    for input_, annot in examples:
        doc_gold_text = ner_model.make_doc(input_)
        gold = GoldParse(doc_gold_text, entities=annot['entities'])
        pred_value = ner_model(input_)
        scorer.score(pred_value, gold)
    return scorer.scores
```

When evaluating the model on the validation data, 

```
evalFold1 = [("""Text …...

spacyModel = spacy.load('en_core_web_lg')
nermodel = spacy.load('myModel')

results = evaluate(nermodel , evalFold1)

```
results produces the following output 

```
{'uas': 0.0,
 'las': 0.0,
 'ents_p': 0.0,
 'ents_r': 0.0,
 'ents_f': 0.0,
 'tags_acc': 0.0,
 'token_acc': 100.0}
```

It also does the same thing when I load the spacy model (en_core_web_lg) and use it to evaluate that same data.

EDIT: this model is the first in 5 part of a 5 fold cross validation set. 

Data is split up like

training folds 2, 3, 4, 5 evaluate on 1 <- only this one gives bad result
training folds 1, 3, 4, 5 evaluate on 2
etc.

So the first model evaluated on the first validation set is giving the incorrect result.  The other 4 are giving normal results (actual values for the P, R and F are like 93%)

**Comment by erotavlas** [comment]
I think I may have found the problem 

In my new training data I found that I was exporting my entities with the character start and end positions different than what spaCy expects (for the end position).

For example

TRAIN_DATA = [("""Hello John Doe""",{'entities':[(6,13,'PERSON')]})

However when I ran nlp method on the same string it outputs the start as 6 and end as 14.

Spacy is going one position past the position of the last character in the entity.
